mlflow/store/hdfs_artifact_repo.py

```diff
@@ -174,11 +183,27 @@ def hdfs_system(host, port):
     connected.close()
 
 
+class HdfsArtifactRepositoryException(Exception):
+    pass
+
+
 def _resolve_connection_params(artifact_uri):
     parsed = urllib.parse.urlparse(artifact_uri)
+    if parsed.scheme == "har":
+        return _parse_har_filesystem(artifact_uri)
     return parsed.hostname, parsed.port, parsed.path
 
 
+def _parse_har_filesystem(artifact_uri):
+    har_extension_index = artifact_uri.find(HAR_EXTENSION)
+    if har_extension_index < 0:
```

Comment:
Wouldn't something like `artifact_uri.endswith(HAR_EXTENSION)` be more robust? .. with this approach I guess that having HAR_EXTENSION appearing even in the middle of the `artifact_uri` would pass this check

Reply:
This function can use a path inside a archive as artifact folder. So .har will be in the middle of the `artifact_uri`

So it is an expected behavior